pkgs/development/lua-modules/overrides.nix

```diff
@@ -428,6 +428,34 @@ with prev;
   });
 
 
+  liblpeg = prev.lpeg.overrideAttrs (oa: {
+    pname = "liblpeg";
+
+    # https://github.com/neovim/neovim/pull/25315
+    # Original author uses gcc for darwin to compile lpeg onto `.so`.
+    # This causes some issue with the common darwin setup to use clang, which
+    # expects `.dylib` for `CMAKE_SHARED_LIBRARY_SUFFIX`
+    buildPhase = ''
+      sed -i makefile -e "s/CC = gcc/CC = clang/"
```

Comment:
this depends on stdenv and while neovim's flake defaults to clang, it's not the case of nixpkgs thus not sure that's needed, it might be autodetected ?